src/servers/src/mysql/helper.rs

```diff
@@ -161,7 +163,28 @@ pub fn convert_value(param: &ParamValue, t: &ConcreteDataType) -> Result<ScalarV
                 String::from_utf8_lossy(b).to_string(),
             ))),
             ConcreteDataType::Binary(_) => Ok(ScalarValue::Binary(Some(b.to_vec()))),
-
+            ConcreteDataType::Timestamp(ts_type) => {
+                let ts = Timestamp::from_str_utc(&String::from_utf8_lossy(b)).map_err(|e| {
+                    error::MysqlValueConversionSnafu {
+                        err_msg: e.to_string(),
+                    }
+                    .build()
+                })?;
+                match ts_type {
+                    TimestampType::Nanosecond(_) => {
+                        Ok(ScalarValue::TimestampNanosecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Microsecond(_) => {
+                        Ok(ScalarValue::TimestampMicrosecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Millisecond(_) => {
+                        Ok(ScalarValue::TimestampMillisecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Second(_) => {
+                        Ok(ScalarValue::TimestampSecond(Some(ts.value()), None))
```

Comment:
I'm afraid we cannot simply use `ts.value` here for different time unit. The input timestamp string will have no unit information so the timestamp we parsed in line 167 will have a fixed unit.

Reply:
Maybe we can use `covert_to()`, for example:

```
let ts = Timestamp::from_str_utc(&String::from_utf8_lossy(b))
                    .map_err(|e| {
                        error::MysqlValueConversionSnafu {
                            err_msg: e.to_string(),
                        }
                        .build()
                    })?
                    .convert_to(ts_type.unit())
                    .ok_or_else(|| {
                        error::MysqlValueConversionSnafu {
                            err_msg: "Overflow when converting timestamp to target unit"
                                .to_string(),
                        }
                        .build()
                    })?;
```

Reply:
sounds good. we can add some unit tests to verify this.